tools/dds/dds-server/lrs-device-manager.cpp

```diff
@@ -0,0 +1,119 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include <algorithm>
+#include <iostream>
+#include "lrs-device-manager.h"
+
+using namespace tools;
+
+class lrs_device_manager::lrs_sensor_streamer
+{
+public:
+    lrs_sensor_streamer() = default;
+    lrs_sensor_streamer( rs2::sensor rs2_sensor,
+                    rs2::stream_profile stream_profile,
+                    std::function< void( const std::string & stream_name, uint8_t * data, int size ) > cb )
+        : _rs2_sensor( rs2_sensor )
+        , _stream_profile( stream_profile )
+        , _frame_callback(std::move( cb ))
+    {
+        _rs2_sensor.open( _stream_profile );  // TODO open required profile!
+        _rs2_sensor.start( [&]( rs2::frame f ) 
+            {
+                _frame_callback( _stream_profile.stream_name(), (uint8_t *)f.get_data(), f.get_data_size() );
```

Comment:
I don't see that the stream name is actually used... why?

Reply:
It is.
at `dds_device_server::publish_frame`
it does:
`stream_name_to_server.at( stream_name )->publish_video_frame( frame, size );`
This tells the device server which stream server callback is relevant